.github/scripts/check-anchors.py

```
@@ -0,0 +1,367 @@
+        return parse_name_status(out.stdout)
+
+    # origin/main unavailable (e.g. shallow direct push): fall back to HEAD~1
+    out = subprocess.run(
+        ['git', 'diff', '--name-status', '-M', 'HEAD~1...HEAD'],
+        capture_output=True, text=True, cwd=root)
+    if out.returncode == 0:
+        return parse_name_status(out.stdout)
+
+    raise RuntimeError(
+        f'git diff failed: origin/main...HEAD rc={out.returncode}; '
+        f'HEAD~1...HEAD rc={out.returncode}: {out.stderr.strip()[:300]}')
```

Ruff (0.16.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)